zebra-chain/src/serialization.rs

```diff
@@ -299,6 +306,23 @@ pub trait ReadZcashExt: io::Read {
 /// Mark all types implementing `Read` as implementing the extension.
 impl<R: io::Read + ?Sized> ReadZcashExt for R {}
 
+/// Helper for deserializing more more succinctly via type inference
+pub trait DeserializeInto {
+    /// Deserialize based on type inference
+    fn deserialize_into<T>(&self) -> Result<T, SerializationError>
+    where
+        T: ZcashDeserialize;
+}
+
+impl DeserializeInto for Vec<u8> {
```

Comment:
The specific types this trait is impl'ed for will probably change as it gets used in more places